src/Microsoft.Windows.CsWin32/templates/SYSTEMTIME.cs

```diff
@@ -0,0 +1,28 @@
+﻿partial struct SYSTEMTIME
+{
+	public static implicit operator global::System.DateTime(SYSTEMTIME sysTime)
+	{
+		if (sysTime.wYear <= 0 || sysTime.wMonth <= 0 || sysTime.wDay <= 0)
+		{
+			Debug.Fail("Incorrect SYSTEMTIME info!");
+			return global::System.DateTime.MinValue;
+		}
+
+		// DateTime gets DayOfWeek automatically
+		return new global::System.DateTime(sysTime.wYear,
+			sysTime.wMonth, sysTime.wDay, sysTime.wHour,
+			sysTime.wMinute, sysTime.wSecond, sysTime.wMilliseconds);
+	}
+
+	public static implicit operator SYSTEMTIME(global::System.DateTime time) => new SYSTEMTIME
+	{
+		wYear = (ushort)time.Year,
+		wMonth = (ushort)time.Month,
+		wDayOfWeek = (ushort)time.DayOfWeek,
+		wDay = (ushort)time.Day,
+		wHour = (ushort)time.Hour,
+		wMinute = (ushort)time.Minute,
+		wSecond = (ushort)time.Second,
+		wMilliseconds = (ushort)time.Millisecond
```

Comment:
All these should be surrounded with a `checked { }` block since we're truncating values. While we may not expect *valid* values to ever exceed `ushort.MaxValue`, this code shouldn't be the means of silently truncating invalid values into something that may appear to be more acceptable.